sftp/sftp.go

```diff
@@ -508,9 +508,9 @@ func (o *Object) Hash(r fs.HashType) (string, error) {
 	err = fs.ErrHashUnsupported
 	var outputBytes []byte
 	if r == fs.HashMD5 {
-		outputBytes, err = session.Output("md5sum " + o.path() + " | awk '{ print $1 }'")
+		outputBytes, err = session.Output("md5sum \"" + o.path() + "\" | awk '{ print $1 }'")
```

Comment:
Better, but still not there yet!

You need something like this...

```
$ gore -autoimport
gore version 0.2.6  :help for help
gore> r := regexp.MustCompile(`[^A-Za-z0-9._/-]`)
(*regexp.Regexp)[^A-Za-z0-9._/-]
gore> r.ReplaceAllString("p*p/hoop-pie's", `\\$1`)
(string)p\\p/hoop-pie\\s
gore> 
```

Stick it in a function called quotemeta is what I would do...